packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -237,22 +282,56 @@ export class DetailsService {
     }
   }
 
+  /**
+   * Assigns reputations to leaders based on fetched reputation records.
+   *
+   * Logic Description:
+   * - A map of leaders is created using their addresses for quick lookup.
+   * - If ordered by REPUTATION:
+   *   - Iterate through reputation records.
+   *   - If a corresponding leader is found, update their reputation.
+   *   - If no leader is found:
+   *     - Log a warning indicating a missing leader for the specific reputation record.
+   *     - This likely indicates an issue where the subgraph did not return the expected leader data.
+   *     - Skip processing the record without substituting a different leader.
+   *   - Return the top `first` leaders based on reputation.
+   * - If not ordered by REPUTATION:
+   *   - Create a map of reputation records for quick lookup.
+   *   - Update each leader's reputation if found, otherwise set it to LOW.
+   * - Return the list of updated leaders.
+   */
   private assignReputationsToLeaders(
     leaders: LeaderDto[],
     reputations: { address: string; reputation: string }[],
-    chainId: ChainId,
-  ) {
+    orderBy: LeadersOrderBy,
+  ): LeaderDto[] {
+    const leaderMap = new Map(leaders.map((l) => [l.address.toLowerCase(), l]));
+
+    if (orderBy === LeadersOrderBy.REPUTATION) {
+      return reputations
+        .map((rep) => {
+          const leader = leaderMap.get(rep.address.toLowerCase());
+          if (!leader) {
+            this.logger.warn(
```

Comment:
We should not be logging this. This is something that might happen if an operator stops running a service and unstakers, removes the role, etc.